sql-odbc/src/PowerBIConnector/OdfeSqlOdbcPBIConnector.pq

```diff
@@ -1,42 +1,214 @@
         Host as (type text meta [
             Documentation.FieldCaption = "Host",
-            Documentation.FieldDescription = "The hostname of the Open Distro For Elasticsearc server.",
+            Documentation.FieldDescription = "The hostname of the Open Distro For Elasticsearch server.",
             Documentation.SampleValues = { "localhost" }
         ]),
-        Port as (type number meta [
+        optional Port as (type number meta [
             Documentation.FieldCaption = "Port",
-            Documentation.FieldDescription = "The port the Open Distro For Elasticsearc server is running on.",
+            Documentation.FieldDescription = "The port of the Open Distro For Elasticsearch server is running on.",
             Documentation.SampleValues = { 9200 }
         ])
     )
     as table meta [
         Documentation.Name = "Open Distro For Elasticsearch"
     ];
 
-OdfeSqlOdbcPBIConnectorImpl = (Host as text, Port as number) as table =>
+OdfeSqlOdbcPBIConnectorImpl = (Host as text, optional Port as number) as table =>
     let
+        Credential = Extension.CurrentCredential(),
+        AuthenticationMode = Credential[AuthenticationKind],
+
+        // Sets connection string properties for authentication.
+        CredentialConnectionString =
+            if AuthenticationMode = "UsernamePassword" then
+                [
+                    Auth = "BASIC",
+                    UID = Credential[Username],
+                    PWD = Credential[Password]
+                ]
+            else if AuthenticationMode = "Key" then
+                [
+                    Auth = "AWS_SIGV4",
+                    Region = Credential[Key]
+                ]
+            else 
+                [
+                    Auth = "NONE"
+                ],
+
+        // Sets connection string properties for encrypted connections.
+        EncryptedConnectionString =
+            if Credential[EncryptConnection] = null or Credential[EncryptConnection] = true then
+                [
+                    UseSSL = 1
+                ]
+            else
+                [
+                    UseSSL = 0
+                ],
+
+        // Set host & port in connection string. 
+        // Do not include port in connection string for aws server connection.
+        Server = 
+            if Port <> null then
+            [
+                Host = Host,
+                Port = Port
+            ]
+            else
+            [
+                Host = Host
+            ],
+
         ConnectionString = [
-            Driver = "ODFE SQL ODBC Driver",
-            Host = Host,
-            Port = Port
+            Driver = "ODFE SQL ODBC Driver"
         ],
-        OdbcDatasource = Odbc.DataSource(ConnectionString)
+
+        SqlCapabilities = Diagnostics.LogValue("SqlCapabilities_Options", [
+            GroupByCapabilities = ODBC[SQL_GB][SQL_GB_COLLATE]
+        ]),
+
+        SQLGetInfo = Diagnostics.LogValue("SQLGetInfo_Options", [
```

Comment:
These are values that we can change at the driver level. Unless those changes break other applications, lets set Capabilities and GetInfo values there instead

Can you remove these lines (SQLCapabilities & SQLGetInfo) and verify if we get the same connection behaviour?

Reply:
Removed SQLCapabilities & SQLGetInfo since all required values are defined at the driver level